pkg/apis/config/v1beta1/types.go

```diff
@@ -0,0 +1,108 @@
+/*
+Copyright 2023 The Kubeflow Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1beta1
+
+import (
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// +kubebuilder:object:root=true
+
+// KatibConfig is the katib-config.yaml structure in Katib config.
+type KatibConfig struct {
+	metav1.TypeMeta `json:",inline"`
+
+	RuntimeConfig RuntimeConfig `json:"runtime,omitempty"`
+	InitConfig    InitConfig    `json:"init,omitempty"`
+}
+
+// RuntimeConfig is the runtime structure in Katib config.
+type RuntimeConfig struct {
+	SuggestionConfigs       []SuggestionConfig       `json:"suggestions,omitempty"`
+	EarlyStoppingConfigs    []EarlyStoppingConfig    `json:"earlyStoppings,omitempty"`
+	MetricsCollectorConfigs []MetricsCollectorConfig `json:"metricsCollectorSidecars,omitempty"`
+}
+
+// InitConfig is the YAML init structure in Katib config.
+type InitConfig struct {
+	ControllerConfig ControllerConfig `json:"controller,omitempty"`
+
+	// TODO (tenzen-y): Adding a config for the webhook certs here.
+	// Ref: https://github.com/kubeflow/katib/issues/2149
+}
+
+// ControllerConfig is the controller structure in Katib config.
+type ControllerConfig struct {
+	// ExperimentSuggestionName is the implementation of suggestion interface in experiment controller.
+	// Defaults to 'default'.
+	ExperimentSuggestionName string `json:"experimentSuggestionName,omitempty"`
+	// MetricsAddr is the address the metric endpoint binds to.
+	// Defaults to ':8080'.
+	MetricsAddr string `json:"metricsAddr,omitempty"`
+	// HealthzAddr is the address the healthz endpoint binds to.
+	// Defaults to ':18080'.
+	HealthzAddr string `json:"healthzAddr,omitempty"`
+	// InjectSecurityContext indicates whether inject the securityContext of container[0] in the sidecar.
+	// Defaults to 'false'.
+	InjectSecurityContext *bool `json:"injectSecurityContext,omitempty"`
+	// EnableGRPCProbeInSuggestion indicates whether enable grpc probe in suggestions.
+	// Defaults to 'true'.
+	EnableGRPCProbeInSuggestion *bool `json:"enableGRPCProbeInSuggestion,omitempty"`
+	// TrialResources is the list of resources that can be used as trial template,
+	// in the form: Kind.Version.Group (e.g. TFJob.v1.kubeflow.org)
+	// It doesn't set as a default.
+	TrialResources []string `json:"trialResources,omitempty"`
```

Comment:
Do you set the default value for TrialResources here: https://github.com/kubeflow/katib/blob/6595c5b8e780f86c27f240d33e7917b928bb73cb/pkg/apis/config/v1beta1/defaults.go#L24 ?

Reply:
Yes, that's right. this comment doesn't correct. Thanks!

Reply:
Done.